pkg/pgmodel/ingestor/copier.go

```diff
@@ -485,14 +486,30 @@ func insertMetadata(conn pgxconn.PgxConn, reqs []pgmodel.Metadata) (insertedRows
 	return insertedRows, nil
 }
 
-var copierChannelMutex sync.Mutex
+var (
+	copierChannelMutex            sync.Mutex
+	samplesCopierChannelToMonitor chan readRequest
+	// Declare and register here since we need readRequest struct which should not be defined in
+	// pgmodel/metric.
+	ingestorChannelLenCopier = prometheus.NewGaugeFunc(
```

Comment:
I failed to understand why can't we just have one gauge metric `IngestorChannelLen` (defined in ingest.go) and use it for both batcher and copier with different label values? That approach would make it for less code and would not require to have a global var `samplesCopierChannelToMonitor`

Reply:
Similar to what's done here https://github.com/timescale/promscale/pull/1177/files#diff-e157d1ebaa83fd2018edfff00e940023c70044b7df48451a3926d0c36af61826L494

Reply:
The recent commit updates this. But basically, one metric (copier channel_len) needs a callback to be updated when prometheus pings the /metric (hence we need NewGaugeFunc), and the other done for metric batcher will be used irrespective of prometheus calls (so NewGauge will work).